### ISSUE TITLE
List of models/tasks failing ONNX inference

### ISSUE THREAD

**Issue by dwyatte** [huggingface/transformers#19357]
### System Info

- `transformers` version: 4.23.0.dev0
- Platform: Darwin-21.6.0-x86_64-i386-64bit
- Python version: 3.7.12
- Huggingface_hub version: 0.10.0
- PyTorch version (GPU?): 1.11.0 (False)
- Tensorflow version (GPU?): 2.9.1 (False)
- Flax version (CPU?/GPU?/TPU?): 0.5.2 (cpu)
- Jax version: 0.3.6
- JaxLib version: 0.3.5
- Using GPU in script?: NA
- Using distributed or parallel set-up in script?: NA

### Who can help?

@lewtun

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

This issue was surfaced in https://github.com/huggingface/transformers/pull/19255 (see also https://github.com/huggingface/transformers/issues/19320)

Export any of the following model/task/framework parameterizations to ONNX:

- [ ] ("deberta-v2", "question-answering", "pt"),
- [ ] ("deberta-v2", "multiple-choice", "pt"),
- [ ] ("roformer", "multiple-choice", "pt"),
- [ ] ("groupvit", "default", "pt"),
- [ ] ("perceiver", "masked-lm", "pt"),
- [ ] ("perceiver", "sequence-classification", "pt"),
- [ ] ("perceiver", "image-classification", "pt"),
- [ ] ("bert", "multiple-choice", "tf"),
- [ ] ("camembert", "multiple-choice", "tf"),
- [ ] ("roberta", "multiple-choice", "tf"),

Errors are currently not detected at export time (although see https://github.com/huggingface/transformers/pull/19255). However, running inference on these exported models with any shape other than what they were validated with will fail.

### Expected behavior

* Errors are raised during ONNX export
* Inference runs as expected
